rudder-web/src/main/scala/com/normation/rudder/web/rest/technique/TechniqueAPIService6.scala

```diff
@@ -0,0 +1,137 @@
+*/
+
+package com.normation.rudder.web.rest.technique
+
+import com.normation.cfclerk.domain._
+import com.normation.eventlog.EventActor
+import com.normation.eventlog.ModificationId
+import com.normation.rudder.batch.AsyncDeploymentAgent
+import com.normation.rudder.batch.AutomaticStartDeployment
+import com.normation.rudder.domain.RudderLDAPConstants
+import com.normation.rudder.domain.policies.ActiveTechnique
+import com.normation.rudder.domain.policies.ChangeRequestDirectiveDiff
+import com.normation.rudder.domain.policies.DeleteDirectiveDiff
+import com.normation.rudder.domain.policies.Directive
+import com.normation.rudder.domain.policies.DirectiveId
+import com.normation.rudder.domain.policies.ModifyToDirectiveDiff
+import com.normation.rudder.domain.policies.SectionVal
+import com.normation.rudder.domain.workflows.ChangeRequestId
+import com.normation.rudder.repository.RoDirectiveRepository
+import com.normation.rudder.repository.WoDirectiveRepository
+import com.normation.rudder.services.workflows.ChangeRequestService
+import com.normation.rudder.services.workflows.WorkflowService
+import com.normation.rudder.web.model.DirectiveEditor
+import com.normation.rudder.web.rest.RestUtils
+import com.normation.rudder.web.rest.RestUtils.getActor
+import com.normation.rudder.web.rest.RestUtils.toJsonError
+import com.normation.rudder.web.rest.RestUtils.toJsonResponse
+import com.normation.rudder.web.services.DirectiveEditorService
+import com.normation.rudder.web.rest.RestExtractorService
+import com.normation.utils.Control._
+import com.normation.utils.StringUuidGenerator
+import net.liftweb.common._
+import net.liftweb.http.Req
+import net.liftweb.json.JArray
+import net.liftweb.json.JValue
+import net.liftweb.json.JsonDSL._
+import com.normation.rudder.web.rest.RestDataSerializer
+import com.normation.rudder.domain.workflows.ChangeRequestId
+import com.normation.cfclerk.domain.TechniqueId
+import com.normation.cfclerk.services.TechniqueRepository
+import com.normation.utils.Control._
+import scala.collection.SortedMap
+import net.liftweb.json.JsonAST.JValue
+
+case class TechniqueAPIService6 (
+    readDirective        : RoDirectiveRepository
+  , restDataSerializer   : RestDataSerializer
+  , techniqueRepository  : TechniqueRepository
+  ) extends Loggable {
+
+  def serialize(technique : Technique, directive:Directive) = restDataSerializer.serializeDirective(technique, directive, None)
+
+  def listTechniques : Box[JValue] = {
+    for {
+      lib <- readDirective.getFullDirectiveLibrary()
+      activeTechniques = lib.allActiveTechniques.values.toSeq
+
+      serialized = activeTechniques.map(restDataSerializer.serializeTechnique)
+    } yield {
+      serialized
+    }
+  }
+
+  def listDirectives (techniqueName : TechniqueName, wantedVersions: Option[List[TechniqueVersion]])  : Box[JValue] = {
+    def serializeDirectives (directives : Seq[Directive], techniques : SortedMap[TechniqueVersion, Technique]) = {
```

Comment:
I don't like having closed parameters used in subfunction. It's harder to refactor, and closure leaks all around in scala. Could you add the wantedVersions parameter to parameters of serializeDirectives?